Feature ptech 364 build scripts

justclint opened the pull request:
### JIRA
https://prometprojects.atlassian.net/browse/PTECH-364

### Local/dev deploy instructions.

1. cd into your project directory.

2. Download the production sites DB to your projects root (not docroot) directory as production-db.sql.

3. Start docker containers.
`docker-compose up -d`

4. Go into Docker container.
`docker-compose exec php /bin/bash`
`cd /var/www/web`

5. Run build script.
`../scripts/drupal-update.sh`

### Known issues
This is the output I pasted in hipchat related to "Drush en or dis" commands. Everything seems to run fine but this output at the end of command being run is not familiar. Again, they are not errors but just odd to see. 

> Disabling modules.
> Disabling modules for development.
> Starting qualitynm_backend_1
> Executing: mysql --defaults-extra-file=/tmp/drush_dxlHfP --database=drupal --host=db --silent  < /tmp/drush_EXv5l7
> Executing: mysql --defaults-extra-file=/tmp/drush_Gb365e --database=drupal --host=db --silent  < /tmp/drush_mNbTax
> Loading release_info engine.                                                                                                                                                                                                          [notice]
> Loading version_control engine.                                                                                                                                                                                                       [notice]
> Loading package_handler engine.                                                                                                                                                                                                       [notice]
> Executing: wget --version
> There were no extensions that could be disabled.                                                                                                                                                                                   [ok]
> Displaying Druplicon for "pm-disable" command.                                                                                                                                                                                        [notice]
> Command dispatch complete

justclint committed:
PTECH-364: Added environment varialbe to docker-compose.yml.

justclint committed:
PTECH-364: Added scripts directory.

- Includes build scripts and script resource files.

justclint committed:
PTECH-364: Removed previously added modules from enable/disable list.

- Just left common modules related to dev.

justclint committed:
PTECH-364: Updated README file.

Luukyb commented on README.md:
````diff
@@ -20,14 +20,22 @@ specific containers to use this project.
+Add your env variable for dev.
+```sh
+$ cd to project root
+$ export BUILD_ENV=dev
````
Do we need this since the variable is in the .env file?

justclint replied:
I started testing this again and I got the variable passed to the container with the .env file. So that part is working now. But cant seem to get the variable from the container when trying to access from outside.  By this I mean for example I can go into the container and echo the variable:
```
promet@Promets-MBP.attlocal.net: $ docker-compose exec php /bin/bash
root@a41c9229fb88:/var/www/web# echo $BUILD_ENV                                                                                                                                                                                              
dev
```

But when trying to access this variable with the build script the variable is always undefined. So for example if in the build script (or directly from the cmd as well) I do:
`docker-compose run --rm php echo $BUILD_ENV`
It will return this:
```
promet@Promets-MBP.attlocal.net: $ ../scripts/drupal-update.sh 
WARNING: The BUILD_ENV variable is not set. Defaulting to a blank string.
Starting simpledockerdrupal_backend_1
```

One work around I found is just taking the variable from the .env file but Im not sure this is what we really want as I suspect we want to read the variable directly from the container. So for example Im able to do this and it does work:
```
source ../.env
echo $BUILD_ENV
```
which outputs:
```
promet@Promets-MBP.attlocal.net: $ ../scripts/drupal-update.sh
dev
```

Any idea what might be the issue?

Luukyb replied:
Hi @justclint,
I think I understand your issue now.
You are running the deployment script from your local, not from the container. This is why you have things like 'drush="docker-compose run --rm php drush"' in your drupal-update.sh file.
What I had in mind was more simple. If you are familar with Promet's current framework, we use the script inside vagrant. I would do the same with Docker, the script is executed inside the container, not your local. This is how you make accessing all the environment variables and other containers easy :)
You would start the script with: 
`docker-compose run --rm php ./../scripts/drupal-update.sh`
And that's it. The script should have access to everything it needs, because it's executed inside the container. Do you see any potential issues compared to what you were doing? Could you give it a try and update your script?
Let me know,
Thanks!
Luc

justclint committed:
PTECH-364: Moved .env file.

- Moved to the .env to same directory as docker-compose.yml file.

justclint commented:
Hey Luc, Im using this to slaughter another nmh site and one thing Im running into on the dev build is a warning regarding devel not being able to download firePHP due to not having svn installed.

```
devel was enabled successfully.                                                                                                                                                                                                    [ok]
devel defines the following permissions: access devel information, execute php code, switch users
Executing: svn export http://firephp.googlecode.com/svn/branches/Library-FirePHPCore-0.3 /var/www/web/sites/all/modules/devel/FirePHPCore
Drush was unable to export FirePHP to /var/www/web/sites/all/modules/devel/FirePHPCore.                                                                                                                                            [warning]
Command dispatch complete  
```                                             

Ive tried adding a check to install svn if not installed but keep getting 
`Unable to locate package subversion.  `                 

Im guessing this is a docker thing since Ive tested apt-get install (and update) with other items and get the same result. 

Per googling this issue its just a warning and not much of a concern but thought Id point this out in case you have any direction on how to handle this.

Just let me know.
Thanks!

justclint committed:
PTECH-364: Made drush command exacutable within container.

justclint committed:
PTECH-364: Added registry_rebuild to docker file.

justclint committed:
PTECH-364: Updated README file.

justclint committed:
PTECH-364: Updated README file.

- Added the instructions to go into the docker container prior to running script.

marychris commented on scripts/drupal-update.sh:
```diff
@@ -0,0 +1,164 @@
+  if [ ! -e "sites/default/settings.local.php" ]
+  then
+    echo "Copying over default datbase settings."
+    cp -v "../scripts/settings.local.dist.php" "sites/default/settings.local.php"
```
This did not work for me, here is the error message (this is a brand new copy / paste of Drupal 8 core):
Copying over default database settings.
'../scripts/settings.local.dist.php' -> 'sites/default/settings.local.php'
cp: cannot create regular file 'sites/default/settings.local.php': Permission denied

Luukyb replied:
I think that anyway this should not be done on update.sh. Update.sh is something we re-run on every deployment. For every deployment, we don't want to replace our local settings file. We would do that only when installing, the first time, which should be scripts/drupal-install.sh.

marychris commented on README.md:
````diff
@@ -20,14 +20,22 @@ specific containers to use this project.
 ```sh
-$  docker-compose run --rm php ./../vendor/bin/drush site-install
+$ cd /docroot
+$ ../scripts/drupal-update.sh
````
I had to use $ docker-compose run --rm php ../scripts/drupal-update.sh instead of just $ ../scripts/drupal-update.sh.

Luukyb commented on scripts/drupal-update.sh:
```diff
@@ -0,0 +1,164 @@
+    echo ""
+  fi
+
+  # Install production backup db.
```
I think this should not be done on update.sh. Update.sh is something we re-run on every deployment. For every deployment, we don't want to replace the db. We would do that only when setting up docker for the first time, which should be scripts/drupal-install.sh.

marychris commented on scripts/drupal-update.sh:
```diff
@@ -0,0 +1,164 @@
+    exit
+  fi
+
+  echo "Changeing Temporary Directory"
```
Misspelled 'Changing' :)

marychris commented on scripts/drupal-update.sh:
```diff
@@ -0,0 +1,164 @@
+  fi
+
+  echo "Changeing Temporary Directory"
+  $drush vset file_temporary_path /tmp
```
This doesn't work on D8 :( 

Error: 
Changeing Temporary Directory
Command variable-set requires Drupal core version 6 or 7 to run.                          [error]
The drush command 'vset file_temporary_path /tmp' could not be executed.                  [error]

Luukyb replied:
I think this should not be done on update.sh. Update.sh is something we re-run on every deployment, and on servers. This is very dangerous to change this kind of variables on servers.
